soccer/gameplay/plays/restarts/their_shoot_out.py

```diff
@@ -0,0 +1,149 @@
+import standard_play
+import behavior
+import main
+import robocup
+import play
+import enum
+import skills.move
+import constants
+import evaluation
+
+class TheirShootOut(play.Play):
+	class State(enum.Enum):
+		#where the goalie should start before the game begins
+		starting = 0
+		#prevent the shooting robot from chipping of the robot and shooting a straight shot
+		block = 1
+		#go recieve the ball
+		capture = 2
+
+	def __init__(self):
+		super().__init__(continuous=True)
+
+		for state in TheirShootOut.State :
+			self.add_state(state, behavior.Behavior.State.running)
+
+		self.add_transition(behavior.Behavior.State.start,
+							TheirShootOut.State.starting, lambda: True,
+							'immediately')
+
+		self.add_transition(TheirShootOut.State.starting, 
+							TheirShootOut.State.block, 
+							lambda: main.game_state().is_playing(),
+							'block')
+
+		#go capture if enemy robot doesn't have the ball or is within chip range
+		self.add_transition(TheirShootOut.State.block,
+							TheirShootOut.State.capture, lambda: not self.has_ball() or self.in_chip_distance(),
+							'capture')
+
+		#go back to blocking if the robot reclaims control of the ball and is outside chip range.
+		self.add_transition(TheirShootOut.State.capture,
+							TheirShootOut.State.block, lambda: self.has_ball() and not self.in_chip_distance(),
+							'reblock')
+
+		self.block_percentage = .50
+		self.chip_distance = 3.0
+		self.ball_lost_distance = 0.5
+
+	def on_enter_starting(self):
+		#move the goalie into the goal for the shoot out
+		start_point = constants.Field.OurGoalSegment.center()
+		self.add_subbehavior(skills.move.Move(start_point), 'start')
+
+	def on_exit_starting(self):
+		self.remove_all_subbehaviors()
+
+	def on_enter_block(self):
+		#find ideal point to move too and move there
+		move_to_point = self.get_transition_point()
+		self.blocker = skills.move.Move(move_to_point)
+		self.add_subbehavior(self.blocker, 'block')
+
+	def execute_block(self): 
+		#update point
+		self.blocker.pos = self.get_transition_point()
+		
+	def on_exit_block(self):
+		self.remove_all_subbehaviors()
+
+	def on_enter_capture(self):
+		#capture ball
+		self.add_subbehavior(skills.capture.Capture(),'capture')
+
+	def on_exit_capture(self):
+		self.remove_all_subbehaviors()
+
+	@classmethod
+	def score (cls):
+		gs = main.game_state()
+		return 0 if gs.is_penalty_shootout() and gs.is_their_penalty() else float("inf")
+
+	@classmethod
+	def is_restart (cls):
+		return False
+
+	@classmethod
+	def handles_goalie(cls):
+		return True
+
+	# test all robots in their fleet to see if one has the ball
+	def has_ball(self):
+		closeIndicator = False
+		for r in main.their_robots():
+			closeIndicator = closeIndicator or robocup.Segment(r.pos, main.ball().pos).length() < self.ball_lost_distance
```

Comment:
The standard thing is points and .mag here but this should works

Reply:
Switched to .mag